service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java

```diff
@@ -142,6 +143,24 @@ private void retry(RequestContext requestContext, Handler<RestResponse> response
               return;
             }
 
+            // no need to generate warn log when schema or service not exist
+            if (HttpStatusClass.CLIENT_ERROR.equals(HttpStatusClass.valueOf(response.statusCode()))) {
+              try {
+                @SuppressWarnings("unchecked")
+                String errorCode = JsonUtils.readValue(bodyBuffer.getBytes(), Map.class).get("errorCode").toString();
+                // 400012: ErrServiceNotExists
+                // 400016: ErrSchemaNotExists
+                if (errorCode.equals("400012") || errorCode.equals("400016")) {
+                  countDownLatch.countDown();
+                  return;
+                }
+              } catch (IOException e) {
+                LOGGER.warn("read value failed from buffer {}", bodyBuffer.toString());
+              } catch (NullPointerException e) {
+                LOGGER.warn("error code not included in buffer {}", bodyBuffer.toString());
```

Comment:
Check  errorCode is null is better, never throw NPE and catch

Reply:
Modified, now check errorCode before using it